include/workaround.h

```diff
@@ -51,14 +51,33 @@
 #define CELERITY_DETAIL_NO_UNIQUE_ADDRESS
 #endif
 
-#if CELERITY_DETAIL_ENABLE_DEBUG && !defined(__SYCL_DEVICE_ONLY__)
-#define CELERITY_DETAIL_ASSERT_ON_HOST(...) assert(__VA_ARGS__)
-#else
-#define CELERITY_DETAIL_ASSERT_ON_HOST(...)
-#endif
-
 #if CELERITY_ACCESSOR_BOUNDARY_CHECK
 #define CELERITY_DETAIL_IF_ACCESSOR_BOUNDARY_CHECK(...) __VA_ARGS__
 #else
 #define CELERITY_DETAIL_IF_ACCESSOR_BOUNDARY_CHECK(...)
 #endif
+
```

Comment:
I think a short comment about the bigger picture of the how and why of this PR would be nice, and probably the best place to put it would be here before the block introducing the associated preprocessor machinery?

Reply:
Added some doc comments.